### PR TITLE
invoice struct to wallet for mints and melts

### PR DESCRIPTION
**Pull request by elnosh**
add new invoice struct for wallet. Similar to previous struct that was using but with other extra fields: `TransactionType` (Mint or Melt) and `SettledAt`.

### CODE REVIEW

**Comment by elnosh** [comment]
@rolznz I think this should help with https://github.com/elnosh/gonuts/issues/8 and https://github.com/elnosh/gonuts/issues/10. Invoices being saved now include both mints and melts. Also, added the `SettledAt` field. Let me know if this works for you :)

File: wallet/storage/storage.go [hunk 1]
```diff
@@ -3,17 +3,49 @@ package storage
...
 	GetKeysets() crypto.KeysetsMap
-	SaveInvoice(lightning.Invoice) error
-	GetInvoice(string) *lightning.Invoice
-	GetInvoices() []lightning.Invoice
+	SaveInvoice(Invoice) error
+	GetInvoice(string) *Invoice
+	GetInvoices() []Invoice
+}
+
+type Invoice struct {
```
Comment (rolznz) [inline wallet/storage/storage.go:38]: Is it possible to store fee paid for outgoing payments?

File: wallet/wallet.go [hunk 2]
```diff
@@ -214,12 +214,16 @@ func (w *Wallet) RequestMint(amount uint64) (*nut04.PostMintQuoteBolt11Response,
 		return nil, fmt.Errorf("error decoding bolt11 invoice: %v", err)
 	}
 
-	invoice := lightning.Invoice{
-		Id:             mintResponse.Quote,
-		PaymentRequest: mintResponse.Request,
-		PaymentHash:    bolt11.PaymentHash,
-		Amount:         amount,
-		Expiry:         mintResponse.Expiry,
+	invoice := storage.Invoice{
```
Comment (rolznz) [inline wallet/wallet.go:217]: Is it possible to pass a description or description hash when requesting a mint?

**Comment by rolznz** [comment]
@elnosh looks great! I added a couple of comments but they are just optional improvements (I'm not sure if Cashu supports them or not)

File: wallet/wallet.go (hunk 2 above)
Comment (rolznz) [reply wallet/wallet.go:217]: Ok, no problem. Thanks!

**Pull request merged**